ports/node-api/CMakeLists.txt

```
@@ -0,0 +1,15 @@
+project(nodelib)
+
+add_custom_target(nodelib ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/node.lib)
+add_custom_command(
+  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/node.lib
+  COMMAND ${NODEJS_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/generateNodeLibDef.js
+  COMMAND ${CMAKE_AR}
+          /def:${CMAKE_CURRENT_BINARY_DIR}/node.def
+          /out:${CMAKE_CURRENT_BINARY_DIR}/node.lib
+          ${CMAKE_STATIC_LINKER_FLAGS}
```

This also needs to also include the configuration specific flags:
```suggestion
          ${CMAKE_STATIC_LINKER_FLAGS}
          $<$<CONFIG:Debug>:${CMAKE_STATIC_LINKER_FLAGS_DEBUG}>
          $<$<CONFIG:Release>:${CMAKE_STATIC_LINKER_FLAGS_RELEASE}>
```